### PR TITLE
55403 - Fix image paths returned from `wp_crop_image`

### PR DESCRIPTION
**Pull request by lipemat**
Version 5.8.0 introduced a new image_editor_output_format filter which
allows you to change the output format (extension) of generated image
sizes. If the generated image extension is changed, wp_crop_images must
honor the change and return the correct extension.

Fixes: #55403

Includes: Tests

Trac ticket: https://core.trac.wordpress.org/ticket/55403

### CODE REVIEW

File: tests/phpunit/tests/image/functions.php [hunk 1]
```diff
@@ -343,6 +343,25 @@ private function get_image_editor_engine_classes() {
 		return $classes;
 	}
 
+
+	public function test_wp_crop_image_extension() {
```
Comment (peterwilsoncc) [inline tests/phpunit/tests/image/functions.php]:
```suggestion
	/*
	 * @ticket 55403
	 */
	public function test_wp_crop_image_extension() {
```

File: tests/phpunit/tests/image/functions.php [hunk 2]
```diff
@@ -343,6 +343,25 @@ private function get_image_editor_engine_classes() {
...
+			100,
+			100,
+			100,
+			100
+		);
+		$this->assertNotWPError( $file );
+		$this->assertFileExists( $file );
+
+		unlink( $file );
+		remove_filter( 'image_editor_output_format', array( $this, 'filter_image_editor_output_format' ) );
```
Comment (peterwilsoncc) [inline tests/phpunit/tests/image/functions.php]:
```suggestion
```

Filters are automatically removed at the end of tests so you don't need to worry about it.

**Review by peterwilsoncc — COMMENTED** [review]
I've added a couple of inline suggestions to the tests.

File: tests/phpunit/tests/image/functions.php [hunk 3]
```diff
@@ -659,6 +680,12 @@ public function test_fallback_intermediate_image_sizes() {
 		}
 	}
 
+
+	public function filter_image_editor_output_format() {
+		return array_fill_keys( array( 'image/jpg', 'image/jpeg', 'image/png' ), 'image/webp' );
+	}
+
+
```
Comment (adamsilverstein) [inline tests/phpunit/tests/image/functions.php]: extra empty line here

**Comment by peterwilsoncc** [comment]
Thanks for making the code changes.

I made a typo in my suggestion yesterday 🤦 so have pushed a fix to your branch. I merged in trunk so I can ensure the tests are still passing before committing.

**Comment by peterwilsoncc** [comment]
Merged in https://core.trac.wordpress.org/changeset/53292

**Pull request closed without merging**